src/Compilers/VisualBasic/Portable/Parser/ParseConditional.vb

```diff
@@ -460,11 +477,50 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
             Return SyntaxFactory.ReferenceDirectiveTrivia(hashToken, referenceKeyword, file)
         End Function
 
-        Private Shared Function ParseBadDirective(hashToken As PunctuationSyntax) As BadDirectiveTriviaSyntax
+        Private Function ParseShebangDirective(hashToken As PunctuationSyntax) As ShebangDirectiveTriviaSyntax
+            Dim builder = PooledStringBuilder.GetInstance()
+            Debug.Assert(CurrentToken.Kind = SyntaxKind.ExclamationToken,
+                         NameOf(ParseShebangDirective) & " called with wrong token")
+            Dim exclamationToken = DirectCast(CurrentToken, PunctuationSyntax)
+            GetNextToken()
+
+            ' Eat tokens until the end of line.
+            While CurrentToken.Kind <> SyntaxKind.EndOfFileToken AndAlso
+                  CurrentToken.Kind <> SyntaxKind.StatementTerminatorToken
+                builder.Builder.Append(CurrentToken.ToFullString())
+                GetNextToken()
+            End While
+
+
+            exclamationToken = exclamationToken.AddTrailingTrivia(SyntaxFactory.CommentTrivia(builder.ToStringAndFree()))
```

Comment:
Also, I think we will want to cons up a zero length "EndOfDirectiveToken" (similar to C#).  I'm not sure if there's already a similar token in VB that could be repurposed.

The reason is that I don't think we want the newline to be part of the #! "message", so the newline will need to get attached to some new token, or alternatively (if we don't want to add a new token), the newline could get attached as leading trivia on the next token.